openslides_backend/action/actions/speaker/mixins.py

```diff
@@ -0,0 +1,38 @@
+from typing import Any, Dict, Optional
+
+from ....permissions.permission_helper import has_perm
+from ....permissions.permissions import Permissions
+from ....shared.exceptions import ActionException
+from ....shared.patterns import Collection, FullQualifiedId
+from ...action import Action
+
+
+class CheckSpeechState(Action):
+    def check_speech_state(
```

Comment:
I would suggest to move the whole validation code from `speaker.update` into the mixin. The way it currently is you still have a lot of duplicated code (fetching the settings & permissions). If you move everything in here and give the db instance of speaker as an additional param (where you give `{}`) for the create case, you can deduplicate even more code.

Reply:
Add the speaker param to the check_speech_state and move the speaker.update code into the mixin.